stripe-core/src/main/java/com/stripe/android/core/exception/InvalidResponseException.kt

```diff
@@ -0,0 +1,24 @@
+package com.stripe.android.core.exception
+
+import androidx.annotation.RestrictTo
+import com.stripe.android.core.StripeError
+
+/**
+ * A [StripeException] indicating that invalid parameters were used in a response. E.g when the
+ * response contains a null field that shouldn't be null, or contains an unknown Enum value that's
+ * not defined in the SDK.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class InvalidResponseException(
```

Comment:
Sounds good. I guess we can wrap any serialization / json exception into this within our generic network error handler then.

Reply:
are we going to bubble this up on any sdk callback? we might not want to make it restricted if so.

Reply:
atm it's not exposed in any public API, when we decided to do so happy to remove it:)